code/game/gamemodes/gang/recaller.dm

```diff
@@ -0,0 +1,454 @@
+						outfits -= 1
+			if("ping")
+				ping_gang(usr)
+	attack_self(usr)
+
+
+/obj/item/device/gangtool/proc/ping_gang(mob/user)
+	if(!user)
+		return
+	var/message = stripped_input(user,"Discreetly send a gang-wide message.","Send Message") as null|text
+	if(!message || !can_use(user))
+		return
+	if(user.z > 2)
+		user << "<span class='info'>\icon[src]Error: Station out of range.</span>"
+		return
+	var/list/members = list()
+	members += gang.gangsters
+	members += gang.bosses
+	if(members.len)
+		var/gang_rank = gang.bosses.Find(user.mind)
+		switch(gang_rank)
+			if(1)
+				gang_rank = "Gang Boss"
+			if(2)
+				gang_rank = "1st Lieutenant"
+			if(3)
+				gang_rank = "2nd Lieutenant"
+			if(4)
+				gang_rank = "3rd Lieutenant"
+			else
+				gang_rank = "[gang_rank - 1]th Lieutenant"
+		var/ping = "<span class='danger'><B><i>[gang.name] [gang_rank]</i>: [message]</B></span>"
+		for(var/datum/mind/ganger in members)
+			if(ganger.current && (ganger.current.z <= 2) && (ganger.current.stat == CONSCIOUS))
+				ganger.current << ping
+		for(var/mob/M in dead_mob_list)
+			M << ping
+		log_game("[key_name(user)] Messaged [gang.name] Gang: [message].")
+
+
+/obj/item/device/gangtool/proc/register_device(mob/user)
+	if(gang)	//It's already been registered!
+		return
+	if((promotable && (user.mind in ticker.mode.get_gangsters())) || (user.mind in ticker.mode.get_gang_bosses()))
+		gang = user.mind.gang_datum
+		gang.gangtools += src
+		icon_state = "gangtool-[gang.color]"
+		if(!(user.mind in gang.bosses))
+			ticker.mode.remove_gangster(user.mind, 0, 2)
+			gang.bosses += user.mind
+			user.mind.gang_datum = gang
+			user.mind.special_role = "[gang.name] Gang Lieutenant"
+			gang.add_gang_hud(user.mind)
+			log_game("[key_name(user)] has been promoted to Lieutenant in the [gang.name] Gang")
+			free_pen = 1
+			gang.message_gangtools("[user] has been promoted to Lieutenant.")
+			user << "<FONT size=3 color=red><B>You have been promoted to Lieutenant!</B></FONT>"
+			ticker.mode.forge_gang_objectives(user.mind)
+			ticker.mode.greet_gang(user.mind,0)
+			user << "The <b>Gangtool</b> you registered will allow you to purchase weapons and equipment, and send messages to your gang."
+			user << "Unlike regular gangsters, you may use <b>recruitment pens</b> to add recruits to your gang. Use them on unsuspecting crew members to recruit them. Don't forget to get your one free pen from the gangtool."
+	else
+		usr << "<span class='warning'>ACCESS DENIED: Unauthorized user.</span>"
+
+/obj/item/device/gangtool/proc/recall(mob/user)
+	if(!can_use(user))
+		return 0
+
+	if(recalling)
+		usr << "<span class='warning'>Error: Recall already in progress.</span>"
+		return 0
+
+	gang.message_gangtools("[usr] is attempting to recall the emergency shuttle.")
+	recalling = 1
+	loc << "<span class='info'>\icon[src]Generating shuttle recall order with codes retrieved from last call signal...</span>"
+
+	sleep(rand(100,300))
```

Comment:
This should really be a `spawn()` instead, because if the user does a recall, then closes the UI whilst waiting for it, it'll pop back open.